src/main/java/redis/clients/jedis/JedisShardInfo.java

```diff
@@ -31,7 +32,7 @@ public int getPort() {
   public JedisShardInfo(String host) {
     super(Sharded.DEFAULT_WEIGHT);
```

Comment:
@nykolaslima OK, I can see "else" state. 
We can remove duplication by adding new setup method which is like BinaryJedis.initializeClientFromURI(), but current state is acceptable.